runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtPlugin.java

```diff
@@ -0,0 +1,478 @@
+		}
+		else if (messageNode.getValue().startsWith("The freezing cold attack"))
+		{
+			interruptType = WintertodtInterruptType.SNOWFALL;
+		}
+		else if (messageNode.getValue().startsWith("The brazier is broken and shrapnel"))
+		{
+			interruptType = WintertodtInterruptType.BRAZIER;
+		}
+		else if (messageNode.getValue().startsWith("You have run out of bruma roots"))
+		{
+			interruptType = WintertodtInterruptType.OUT_OF_ROOTS;
+		}
+		else if (messageNode.getValue().startsWith("Your inventory is too full"))
+		{
+			interruptType = WintertodtInterruptType.INVENTORY_FULL;
+		}
+		else if (messageNode.getValue().startsWith("You fix the brazier"))
+		{
+			interruptType = WintertodtInterruptType.FIXED_BRAZIER;
+		}
+		else if (messageNode.getValue().startsWith("You light the brazier"))
+		{
+			interruptType = WintertodtInterruptType.LIT_BRAZIER;
+		}
+		else if (messageNode.getValue().startsWith("The brazier has gone out."))
+		{
+			interruptType = WintertodtInterruptType.BRAZIER_WENT_OUT;
+		}
+
+		if (interruptType == null)
+		{
+			return;
+		}
+
+		boolean wasInterrupted = false;
+		boolean wasDamaged = false;
+		boolean neverNotify = false;
+
+		switch (interruptType)
+		{
+			case COLD:
+			case BRAZIER:
+			case SNOWFALL:
+				wasDamaged = true;
+
+				// Recolor message for damage notification
+				messageNode.setRuneLiteFormatMessage(wrapTextWithColour(messageNode.getValue(), config.damageNotificationColor()));
+				chatMessageManager.update(messageNode);
+				client.refreshChat();
+
+				// all actions except woodcutting are interrupted from damage
+				if (currentActivity != WintertodtActivity.WOODCUTTING)
+				{
+					wasInterrupted = true;
+				}
+
+				break;
+			case INVENTORY_FULL:
+			case OUT_OF_ROOTS:
+			case BRAZIER_WENT_OUT:
+				wasInterrupted = true;
+				break;
+			case LIT_BRAZIER:
+			case FIXED_BRAZIER:
+				wasInterrupted = true;
+				neverNotify = true;
+				break;
+		}
+
+		if (!neverNotify)
+		{
+			boolean shouldNotify = false;
+
+			switch (config.notifyCondition())
+			{
+				case ONLY_WHEN_INTERRUPTED:
+					if (wasInterrupted)
+					{
+						shouldNotify = true;
+					}
+					break;
+				case WHEN_DAMAGED:
+					if (wasDamaged)
+					{
+						shouldNotify = true;
+					}
+					break;
+				case EITHER:
+					shouldNotify = true;
+					break;
+			}
+
+			if (shouldNotify)
+			{
+				notifyInterrupted(interruptType, wasInterrupted);
+			}
+		}
+
+		if (wasInterrupted)
+		{
+			currentActivity = WintertodtActivity.IDLE;
+		}
+	}
+
+	private void notifyInterrupted(WintertodtInterruptType interruptType, boolean wasActivityInterrupted)
+	{
+		final StringBuilder str = new StringBuilder();
+
+		str.append("Wintertodt: ");
+
+		if (wasActivityInterrupted)
+		{
+			str.append(currentActivity.getActionString());
+			str.append(" interrupted! ");
+		}
+
+		str.append(interruptType.getInterruptSourceString());
+
+		String notification = str.toString();
+		log.debug("Sending notification: {}", notification);
```

Comment:
This debug log should totally be part of notifier and not here

Reply:
I was using this to test it since I don't get notifications when focused.